### ISSUE TITLE
Improve xhyve error reporting

### ISSUE THREAD

**Issue by jayunit100** [minishift/minishift#997]
**Problem** 

Looks like my minishift isn't working on a corporate laptop OOTB, I suspect its related to security but not sure yet.  

**Suggested improvement**

Prevent users from needing to know wether "should we be troubleshoot xhyve ?" or "should we be troubleshooting minishift?" .

- https://docs.openshift.org/latest/minishift/getting-started/docker-machine-drivers.html is a 404
- followed: 
```
  brew install docker-machine-driver-xhyve
  sudo chown root:wheel $(brew --prefix)/opt/docker-machine-driver-xhyve/bin/docker-machine-driver-xhyve
  sudo chmod u+s $(brew --prefix)/opt/docker-machine-driver-xhyve/bin/docker-machine-driver-xhyve
```
- ➜ `minishift start` gives an ssh error that is non-descript.  
```
Starting local OpenShift cluster using 'xhyve' hypervisor...
E0606 09:29:00.969383   61421 start.go:176] Error starting the VM: Error configuring authorization on host: Too many retries waiting for SSH to be available.  Last error: Maximum number of retries (60) exceeded. Retrying.
```

I suspect these are all related to xhyve.  Maybe running a smoke test for xyhve inside minishift, and then prompting the user to say `Xyhve is not working correctly on this machine, please install it according to ... and then retry minishift. Thanks!`  would yield a simpler UX.

**Comment by LalatenduMohanty** [comment]
> https://docs.openshift.org/latest/minishift/getting-started/docker-machine-drivers.html is a 404

Right. The file name has changed to https://docs.openshift.org/latest/minishift/getting-started/setting-up-driver-plugin.html

I could not find the broken link as if I start from https://docs.openshift.org/latest/minishift/getting-started/installing.html links are working fine for me.  Can you please point us to the page where you found the broken link?

**Comment by LalatenduMohanty** [comment]
Ideally the steps for Xhyve driver should have worked. Are you still facing the issue? You might want to try the manual steps mentioned in the documentation. 

> I suspect these are all related to xhyve. Maybe running a smoke test for xyhve inside minishift, and then prompting the user to say Xyhve is not working correctly on this machine, please install it according to ... and then retry minishift. Thanks! would yield a simpler UX.

Right, if we can verify that the perquisites are not in place then report, it would improve UX. You  suggestion is similar to https://github.com/minishift/minishift/issues/485 and #966

**Comment by jayunit100** [comment]
Thanks, looking forward to the fixes

**Comment by hferentschik** [comment]
I would argue that this is also covered now by the pull request #1162 for issue #485. @gbraad wdyt?

**Comment by gbraad** [comment]
Most of it, yes... but
`Error starting the VM: Error configuring authorization on host: Too many retries waiting for SSH to be available.  Last error: Maximum number of retries (60) exceeded. Retrying.`

will not be solved, as this is related to the VM getting an IP address. And in this case `libmachine` handles connecting with a connect-retry loop, waiting for the VM to have SSH reachable. however, they never test if the node is reachable, AFTER it got the IP address. Use of a VPN, which sets routes, might cause the VM to be unreachable on the IP address issued.

**Comment by hferentschik** [comment]
I mark this still as resolved though as part of #485. See whether we get more error reports with xhyve with the latest changes. Besides, there is not much we can do with the libmachine code.
